pkg/env/dev.go

```diff
@@ -84,30 +84,26 @@ func (d *dev) Listen() (net.Listener, error) {
 	return net.Listen("tcp", "localhost:8443")
 }
 
-func (d *dev) FPAuthorizer(tenantID, resource string) (refreshable.Authorizer, error) {
-	oauthConfig, err := adal.NewOAuthConfig(d.Environment().ActiveDirectoryEndpoint, tenantID)
-	if err != nil {
-		return nil, err
-	}
-
+func (d *dev) FPAuthorizer(tenantID string, scopes ...string) (autorest.Authorizer, error) {
 	fpPrivateKey, fpCertificates := d.fpCertificateRefresher.GetCertificates()
 
-	sp, err := adal.NewServicePrincipalTokenFromCertificate(*oauthConfig, d.fpClientID, fpCertificates[0], fpPrivateKey, resource)
+	options := d.Environment().ClientCertificateCredentialOptions()
+	tokenCredential, err := azidentity.NewClientCertificateCredential(
+		tenantID, d.fpClientID, fpCertificates, fpPrivateKey, options)
```

Comment:
It looks like old functionality only passed the first certificate instead of the whole array. Do we know this only contains the 1 cert we want on this, or are we potentially expanding the privilege here?

Reply:
Don't think I considered that.  It was just "function wants an array, I have an array, here ya go!"

I suspect there's only one certificate present but I'll try to track this down to make sure.

Reply:
So I kinda side-stepped your question in a "fixup" commit below by rewriting `FPAuthorizer` to get its token credential from the recently-added `FPNewClientCertificateCredential` method (the two methods were _almost_ identical).

But `FPNewClientCertificateCredential` is also passing the full certificate array returned by the certificate refresher.

Did some debugging of E2E failures and in the course of that discovered in the `e2e-eastus-svc` keyvault, the `rp-firstparty` certificate bundle that we're fetching here consists of 3 certificates.  I don't have permission to examine them in the Azure portal, but I assume the other two are just certificate authorities.

I can look into that further if you want.  I guess I could just as easily change it to `fpCertificates[:1]` but I'm not yet convinced there's a risk.